### ISSUE TITLE
setIn not existent array index inside Map raise exception

### ISSUE THREAD

**Issue by rogeliotg** [immutable-js/immutable-js#1428]
My problem is, using inmutable in this structure:

```
myMap = Map({a: [], b: []});
myMAp.setIn(['a', 0], 'v');
```
this is getting an exception

```
immutable.js:870 Uncaught (in promise) Error: invalid keyPath
    at invariant (immutable.js:870)
    at updateInDeepMap (immutable.js:1974)
    at updateInDeepMap (immutable.js:1980)
    at Map.updateIn (immutable.js:1278)
    at Map.setIn (immutable.js:1256)
```
I know I can check if such index exist, initialize if not, then try the update, but maybe a feature request is feasible so inmutable let do this on one step. I don't see any motive for inmutable raising an exception on this use case. I'm reading the library spec on http://untangled.io/immutable-js-all-the-examples-youll-ever-need-to-get-set-and-delete-data-from-lists/ And what I propose is what is normal on lists:

If an index doesn’t exist anywhere in the keyPath, then List.setin() will do one of the following:

If the index preceding the bad index references a List, then the bad index is trying to insert a value at a place outside of the bounds of this list. Usual rules apply, and the List expands, the value is placed at the end of the expanded List, and nulls are inserted in between. Try it for yourself in the example below, by replacing the last line with avengersList.setIn([1, 1, 1, 3], 'scarletWitch'); If the index preceding the bad index is a value (i.e. not a List or Map), then the bad index is trying to insert a value into a data type that has no concept of an index; accordingly, an Invalid keyPath error will be thrown. Try it for yourself in the example below, by replacing the last line with avengersList.setIn([0, 1], 'scarletWitch');. This tries to insert ‘scarletWitch’ into the String ‘ironMan’, and that’s never going to work!

**Comment by jdeniau** [comment]
Thank you for your bug report. The immutable-js oss fork will soon be merged and a fix for this issue has been included in #1833. Once this PR is merged, this issue will be resolved in the main branch. We will then do our best to to release the `4.0.0` version.

Commit reference: https://github.com/immutable-js/immutable-js/pull/1833/commits/6cb40e4ee224ae523d0d92078f64c85109ac368e

**Comment by jdeniau** [comment]
Hi!
The 4.0.0-rc.14 version has been released and should fix this issue.
Feel free to test it and re-open this issue if the issue is still present.
Thank you!
